url71665d7d68_branscha_lib_scripty_tgz-pJ8-com_sdicons_scripty_ScriptyAutoReplJ8/src/com/sdicons/scripty/parser/Eval2.java

```diff
@@ -0,0 +1,1640 @@
+            currstack = currstack.getPrevStack();
+        }
+
+        // Notify listeners that we reached a result.
+        fireFinishedEval(currstack, lResult);
+        return lResult;
+    }
+
+    protected Object eval2(EvalStack aStack)
+    throws CommandException
+    {
+        while(aStack.hasMoreSteps()) step(aStack);
+        if(aStack.size() == 1 && aStack.top().isEvaluated()) return aStack.top().getResult();
+        else throw new CommandException("Internal eval error, invalid state.", aStack);
+    }
+
+    private static final IFrameHandler ATOMIC_HANDLER = new AtomicHandler();
+    private static final IFrameHandler PAIR_HANDLER = new PairHandler();
+
+    private static final IFrameHandler QUOTE_HANDLER = new QuoteHandler();
+    private static final IFrameHandler IF_HANDLER = new IfHandler();
+    private static final IFrameHandler WHILE_HANDLER = new WhileHandler();
+    private static final IFrameHandler AND_HANDLER = new AndHandler();
+    private static final IFrameHandler OR_HANDLER = new OrHandler();
+    private static final IFrameHandler NOT_HANDLER = new NotHandler();
+    private static final IFrameHandler SET_HANDLER = new SetHandler();
+    private static final IFrameHandler LET_HANDLER = new LetHandler();
+    private static final IFrameHandler GET_HANDLER = new GetHandler();
+    private static final IFrameHandler LAMBDA_HANDLER = new LambdaHandler();
+    private static final IFrameHandler DEFUN_HANDLER = new DefunHandler();
+    private static final IFrameHandler TIMER_HANDLER = new TimerHandler();
+
+    private final MacroHandler MACRO_HANDLER = new MacroHandler();
+    private final StandardHandler STANDARD_HANDLER = new StandardHandler();
+
+    public void setCommandRepo(CommandRepository aRepo)
+    {
+        STANDARD_HANDLER.setCommands(aRepo);
+    }
+
+    public CommandRepository getCommandRepo()
+    {
+        return STANDARD_HANDLER.getCommands();
+    }
+
+    public void setMacroRepo(CommandRepository aRepo)
+    {
+        MACRO_HANDLER.setMacros(aRepo);
+    }
+
+    public CommandRepository getMacroRepo()
+    {
+        return MACRO_HANDLER.getMacros();
+    }
+
+    private static final Map<String, IFrameHandler> HANDLERS = new HashMap<String, IFrameHandler>();
+    static
+    {
+        HANDLERS.put("quote", QUOTE_HANDLER);
+        HANDLERS.put("if", IF_HANDLER);
+        HANDLERS.put("while", WHILE_HANDLER);
+        HANDLERS.put("and", AND_HANDLER);
+        HANDLERS.put("or", OR_HANDLER);
+        HANDLERS.put("not", NOT_HANDLER);
+        HANDLERS.put("set", SET_HANDLER);
+        HANDLERS.put("defvar", SET_HANDLER);
+        HANDLERS.put("let", LET_HANDLER);
+        HANDLERS.put("let*", LET_HANDLER);
+        HANDLERS.put("get", GET_HANDLER);
+        HANDLERS.put("lambda", LAMBDA_HANDLER);
+        HANDLERS.put("defun", DEFUN_HANDLER);
+        HANDLERS.put("timer", TIMER_HANDLER);
+    }
+
+    public void step(EvalStack aStack)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.